internal/repository/prisma/dbsqlc/workers.sql

```diff
@@ -88,6 +88,31 @@ WHERE
     "id" = @id::uuid
 RETURNING *;
 
+
+-- name: ResolveWorkerSemaphoreSlots :execrows
+UPDATE "WorkerSemaphore" ws
+SET "slots" = COALESCE(w."maxRuns", 100) - COALESCE(
+    (
+        SELECT COUNT(*)
+        FROM "StepRun" sr
+        WHERE sr."status" IN ('RUNNING', 'ASSIGNED')
+            AND sr."semaphoreReleased" = FALSE
+            AND sr."workerId" = w."id"
+    ), 0
+)
+FROM "Worker" w
+WHERE ws."workerId" = w."id"
+    AND "slots" != COALESCE(w."maxRuns", 100) - COALESCE(
```

Comment:
I think this should be fine for now, its not guaranteed to update on every interval but it will update to the correct value